### ISSUE TITLE
[Patch] Solution for Change Grip/Order menu closing

### ISSUE THREAD

**Issue by agustincampeny** [NathanReeves/BlueCubeMod#2]
Hi!

I'm currently working on making an open source wireless SNES controller for the Switch, based on the ESP32 code that you wrote.

As the code already works really well, I'm trying to fix some things as to improve it.

I've managed to fix the Grip/Order menu closing by simply changing the length of the `report30` array in this chunk:

https://github.com/NathanReeves/BlueCubeMod/blob/3615f8f9732d90824d4daae8e04659d1dd8173f0/Firmware/BlueCubeModv2/main/main.c#L389

What I did was this:

```c
static uint8_t report30[49] = {
    [1] = 0x30,
    [2] = 0x0,
    [3] = 0x80,
    [13] = 0x08
};
```

I will try to notify you if I find other fixes to your code, If you're interested of course.

**Comment by agustincampeny** [comment]
My fix was completely wrong. It definitely worked but because of the incorrect reasons.

The new `report30` array I made wasn't right because I put the first value in spot 1 instead of spot 0 (Matlab user), but this mistake solved the issue (but made the controller unusable). This made me realize that the problem was that the ESP was sending reports while pairing still wasn't ready, and this is what was closing the Grip/Order menu.

The definitive fix was to not send reports until pairing was complete. I modified this bit:

https://github.com/NathanReeves/BlueCubeMod/blob/3615f8f9732d90824d4daae8e04659d1dd8173f0/Firmware/BlueCubeModv2/main/main.c#L423-L430

And changed it for this:

```c
if(timer == 255)
        timer = 0;
    //esp_hid_device_send_report(ESP_HIDD_REPORT_TYPE_INTRDATA, 0xa1, sizeof(report30), report30);

    if(!paired) {
        esp_hid_device_send_report(ESP_HIDD_REPORT_TYPE_INTRDATA, 0xa1, sizeof(dummy), dummy);
        vTaskDelay(100);
    }
    else {
        esp_hid_device_send_report(ESP_HIDD_REPORT_TYPE_INTRDATA, 0xa1, sizeof(report30), report30);
        vTaskDelay(15);
    }
```

Where dummy is a one element array with value 0.
